sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -438,24 +506,61 @@ def update_entity(
                 :dedent: 16
                 :caption: Updating an already existing entity in a Table
         """
+    
+    @overload
+    def update_entity(
+        self,
+        entity: T,
+        mode: UpdateMode = UpdateMode.MERGE,
+        *,
+        etag: Optional[str] = None,
+        match_condition: Optional[MatchConditions] = None,
+        encoder: TableEntityEncoderABC[T],
+        **kwargs
+    ) -> Dict[str, Any]:
+        """Update entity in a table.
+
+        :param entity: The properties for the table entity.
+        :type entity: Custom entity type
+        :param mode: Merge or Replace entity.
+        :type mode: ~azure.data.tables.UpdateMode
+        :keyword etag: Etag of the entity.
+        :paramtype etag: str or None
+        :keyword match_condition: The condition under which to perform the operation.
+            Supported values include: MatchConditions.IfNotModified, MatchConditions.Unconditionally.
+        :paramtype match_condition: ~azure.core.MatchConditions or None
+        :keyword encoder: The encoder used to serialize the outgoing Tables entities.
+        :type encoder: ~azure.data.Tables.TableEntityEncoderABC
+        :return: Dictionary mapping operation metadata returned from the service.
+        :rtype: dict[str, Any]
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+        """
+    
+    @distributed_trace
+    def update_entity(self, *args, **kwargs) -> Dict[str, Any]:
+        entity = args[0]
+        mode = args[1]
+        encoder = kwargs.pop(encoder, DEFAULT_ENCODER)
+        match_condition = kwargs.pop("match_condition", None)
+        etag = kwargs.pop("etag", None)
         if match_condition and not etag and isinstance(entity, TableEntity):
             etag = entity.metadata.get("etag")
         match_condition = _get_match_condition(
             etag=etag, match_condition=match_condition or MatchConditions.Unconditionally
         )
-        entity_copy = _add_entity_properties(entity)
-        partition_key = entity_copy["PartitionKey"]
-        row_key = entity_copy["RowKey"]
+        entity_json = encoder.encode_entity(entity)
+        partition_key = entity_json["PartitionKey"]
+        row_key = entity_json["RowKey"]
```

Comment:
I think you should safelt be able to pop the values on lines 561 and 562 without it being breaking....
```python
partition_key = entity_json.pop("PartitionKey")
row_key = entity_json.pop("RowKey")
```